packages/app/components/stackbox/FrequencyOptionsCard.tsx

```diff
@@ -0,0 +1,132 @@
+"use client";
+
+import { ChangeEvent, useEffect, useState } from "react";
+
+import { twMerge } from "tailwind-merge";
+
+import { FREQUENCY_OPTIONS } from "@/models";
+import { BodyText, RadioButton, TextInput } from "@/ui";
+
+interface FrequencyOptionsCardProps {
+  frequency: FREQUENCY_OPTIONS;
+  setEndDate: (date: Date) => void;
+}
+
+const defaultFrequencyOptions = {
+  [FREQUENCY_OPTIONS.hour]: ["6", "12", "24"],
+  [FREQUENCY_OPTIONS.day]: ["7", "15", "30"],
+  [FREQUENCY_OPTIONS.week]: ["4", "8", "12"],
+  [FREQUENCY_OPTIONS.month]: ["2", "6", "12"],
+};
+const maxCustomFrequencies = {
+  [FREQUENCY_OPTIONS.hour]: 96,
+  [FREQUENCY_OPTIONS.day]: 365,
+  [FREQUENCY_OPTIONS.week]: 52,
+  [FREQUENCY_OPTIONS.month]: 24,
+};
+const postiveIntegerOnly = /^$|^[0-9]+$/;
+
+const getEndDateByDefaultFrequency = (
+  frequency: FREQUENCY_OPTIONS,
+  timeAmount: number
+) => {
+  const dateNow = new Date();
+  const endDateByDefaultFrequency = {
+    [FREQUENCY_OPTIONS.hour]: dateNow.setHours(dateNow.getHours() + timeAmount),
+    [FREQUENCY_OPTIONS.day]: dateNow.setDate(dateNow.getDate() + timeAmount),
+    [FREQUENCY_OPTIONS.week]: dateNow.setDate(
+      dateNow.getDate() + timeAmount * 7
+    ),
+    [FREQUENCY_OPTIONS.month]: dateNow.setMonth(
+      dateNow.getMonth() + timeAmount
+    ),
```

Comment:
Maybe is better to use https://date-fns.org/v4.1.0/docs/add# here. I feel like months are not being set correctly on the UI. Example: selected 12 months, but gives me +15 months

![image](https://github.com/user-attachments/assets/795ee529-7ec2-42e4-88e1-f7234d1bf0ff)

Reply:
I know see that the bug is related to this https://github.com/SwaprHQ/stackly-ui/pull/212/files#r1858630377

But we should use date-fns anyway, which as simpler syntax.